src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/ExpressionHelper.cs

```diff
@@ -100,12 +114,18 @@ public static string GetExpressionText(LambdaExpression expression)
                 nameParts.Pop();
             }
 
+            expressionText = string.Empty;
             if (nameParts.Count > 0)
             {
-                return nameParts.Aggregate((left, right) => left + right).TrimStart('.');
+                expressionText = nameParts.Aggregate((left, right) => left + right).TrimStart('.');
```

Comment:
All this will be optimized in part 2 where we replace the Stack with Stringbuilder